apps/judicial-system/backend/migrations/20210608081018-update-case.js

```diff
@@ -0,0 +1,168 @@
+  accusedNationalId,
+  accusedName,
+  court,
+  requestedValidToDate,
+  isolation,
+  isExtension,
+  previousDecision,
+) {
+  return `Þess er krafist að ${accusedName}, kt. ${
+    accusedNationalId?.length === 10
+      ? `${accusedNationalId.slice(0, 6)}-${accusedNationalId.slice(6)}`
+      : accusedNationalId
+  }, sæti${
+    isExtension && previousDecision === 'ACCEPTING' ? ' áframhaldandi' : ''
+  } ${
+    type === 'CUSTODY' ? 'gæsluvarðhaldi' : 'farbanni'
+  } með úrskurði ${court?.replace('Héraðsdómur', 'Héraðsdóms')}, til ${
+    isValid(requestedValidToDate)
+      ? format(requestedValidToDate, 'PPPPp', { locale: is })
+          ?.replace('dagur,', 'dagsins')
+          ?.replace(' kl.', ', kl.')
+      : ''
+  }${
+    type === 'CUSTODY' && isolation
+      ? ', og verði gert að sæta einangrun á meðan á varðhaldi stendur'
+      : ''
+  }.`
+}
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction((t) =>
+      Promise.all([
+        queryInterface
+          .renameColumn('case', 'other_demands', 'demands', {
+            transaction: t,
+          })
+          .then(() =>
+            queryInterface.sequelize.query(
+              'SELECT "id", "type", "accused_national_id", "accused_name", "court_id", "requested_valid_to_date", "demands", "requested_custody_restrictions", "parent_case_id" \
+               FROM "case"',
+              { transaction: t },
+            ),
+          )
+          .then((res) =>
+            Promise.all(
+              res[0].map(async (c) => {
+                const subRes = await Promise.all([
+                  queryInterface.sequelize.query(
+                    `SELECT "name" \
+                     FROM "institution" \
+                     WHERE "id" = ${c.court_id ? `'${c.court_id}'` : 'NULL'}`,
+                    { transaction: t },
+                  ),
+                  queryInterface.sequelize.query(
+                    `SELECT "decision" \
+                     FROM "case" \
+                     WHERE "id" = ${
+                       c.parent_case_id ? `'${c.parent_case_id}'` : 'NULL'
+                     }`,
+                    { transaction: t },
+                  ),
+                ])
+
+                return {
+                  id: c.id,
+                  type: c.type,
+                  accusedNationalId: c.accused_national_id,
+                  accusedName: c.accused_name,
+                  courtName:
+                    subRes[0][0].length === 1 ? subRes[0][0][0].name : '',
+                  requestedValidToDate: c.requested_valid_to_date,
+                  demands: c.demands,
+                  isolation: c.requested_custody_restrictions?.includes(
+                    'ISOLATION',
+                  ),
+                  isExtension: subRes[1][0].length === 1,
+                  previousDecision:
+                    subRes[1][1].rowCount === 1
+                      ? subRes[1][0][0].decision
```

Comment:
😳

Reply:
I'm just going to take your word for this 🙈

Reply:
Right, there was actually an error there. We cannot easily access rowCount so it is better to check the length of the result. I forgot to change that in one place.